blockchain/interface.go

```diff
@@ -0,0 +1,18 @@
+package blockchain
+
+import (
+	"context"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+type SeededRequest struct {
+	Seed string `json:"seed"`
+}
+
+type Blockchain interface {
+	Start(context.Context) error
+	Seeded(context.Context, interface{}, SeededRequest) ([]byte, error)
```

Comment:
* I am curious why `interface{}`? What does this param represent?
* What does "Seeded" do? The name of this function doesn't seem right; it should read as an action or a command. "seeded" reads as a noun.